pallets/api/src/mock.rs

```diff
@@ -0,0 +1,122 @@
+use frame_support::{
+	derive_impl, parameter_types,
+	traits::{AsEnsureOriginWithArg, ConstU128, ConstU32, Everything},
+};
+use frame_system::{EnsureRoot, EnsureSigned};
+use sp_core::H256;
+use sp_runtime::{
+	traits::{BlakeTwo256, IdentityLookup},
+	BuildStorage,
+};
+
+type Block = frame_system::mocking::MockBlock<Test>;
+pub(crate) type AccountId = u64;
+pub(crate) type AssetId = u32;
+pub(crate) type Balance = u128;
+
+// Configure a mock runtime to test the pallet.
+frame_support::construct_runtime!(
+	pub enum Test
+	{
+		System: frame_system,
+		Assets: pallet_assets::<Instance1>,
+		Balances: pallet_balances,
+		Fungibles: crate::fungibles,
+	}
+);
+
+parameter_types! {
+	pub const BlockHashCount: u64 = 250;
+	pub const SS58Prefix: u8 = 42;
+}
+
+#[derive_impl(frame_system::config_preludes::TestDefaultConfig as frame_system::DefaultConfig)]
+impl frame_system::Config for Test {
```

Comment:
The reason for using those old configurations was because of the old polkadot version we are currently add. I wasn't sure those were already available so I took reference of another project we are building when it was on polkadot 1.7. I suggest we refactor those when doing the upgrades.